src/appleseed/foundation/meta/benchmarks/benchmark_basis.cpp

```diff
@@ -0,0 +1,133 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2010-2013 Francois Beaune, Jupiter Jazz Limited
+// Copyright (c) 2014-2017 Francois Beaune, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.foundation headers.
+#include "foundation/math/basis.h"
+#include "foundation/math/rng/distribution.h"
+#include "foundation/math/rng/mersennetwister.h"
+#include "foundation/utility/benchmark.h"
+
+using namespace foundation;
+
+BENCHMARK_SUITE(Foundation_Math_Basis)
+{
+    const size_t N = 100000;
+
+    template <typename T>
+    struct BuildFixture
+    {
+        Vector<T, 3>  m_normal[N];
+        Basis3<T>     m_result[N];
+
+        BuildFixture()
+        {
+            MersenneTwister rng;
+
+            for (size_t i = 0; i < N; ++i)
+            {
+                m_normal[i] = rand_vector1<Vector<T, 3>>(rng);
```

Comment:
`rand_vector1<Vector<T, 3>>` is only allowed in C++11 and later. In C++03 and earlier, you need to write

    rand_vector1<Vector<T, 3> >